torch/_inductor/autotune_process.py

```diff
@@ -34,58 +34,45 @@ class Pong:
 
 @dataclasses.dataclass
 class TuningProcess:
-    process: Optional[BaseProcess] = None
-    request_queue: Optional["Queue[Any]"] = None
-    response_queue: Optional["Queue[Any]"] = None
+    process: Optional["subprocess.Popen[bytes]"] = None
 
     @staticmethod
-    def process_main(
-        request_queue: "Queue[Any]",
-        response_queue: "Queue[Any]",
-    ) -> None:
-        print("enter child process main")
-        while True:
-            obj = request_queue.get()
+    def process_main() -> None:
+        """
+        Work loop for the benchmarking subprocess.
+        """
+        if DEBUG:
+            print("Entering TuningProcess child main", file=sys.stderr)
+
+        def reply(obj):
+            pickle.dump(obj, sys.stdout.buffer)
+            sys.stdout.flush()
 
+        while True:
+            obj = pickle.load(sys.stdin.buffer)
             if obj is None:
-                break  # None is a sentinel for the child to terminate
+                # None is a sentinel for the child to terminate
+                break
             elif isinstance(obj, Ping):
-                response_queue.put(Pong())
+                reply(Pong())
             elif isinstance(obj, BenchmarkRequest):
-                response_queue.put(obj.benchmark())
+                reply(obj.benchmark())
             else:
                 raise RuntimeError(f"Invalid request type {type(obj)}")
 
-    def valid(self) -> bool:
-        return (
-            self.process is not None
-            and self.request_queue is not None
-            and self.response_queue is not None
-        )
-
-    def clear(self) -> None:
-        self.process = self.request_queue = self.response_queue = None
-
     def initialize(self) -> None:
         """
-        Create child process, request/response queues and do the warm up.
+        Create child process and do the warm up.
         """
-        if self.valid():
+        if self.process is not None:
             return
 
-        # cuda runtime does not work with "fork", use "spawn" to start processes.
-        ctx = multiprocessing.get_context("spawn")
-        request_queue = self.request_queue = ctx.Queue()
-        response_queue = self.response_queue = ctx.Queue()
-
-        process = self.process = ctx.Process(
-            target=self.process_main,
-            args=(
-                self.request_queue,
-                self.response_queue,
-            ),
+        self.process = subprocess.Popen(
+            [sys.executable, "-m", "torch._inductor.autotune_process_entry"],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
```

Comment:
Let me take a step back. Would this be a bit fragile. Any extra print to stdout in child process would mess things up?

Reply:
Within the subprocess, we could capture stdout when we are not explicitly communicating

Reply:
Yes, it would be vulnerable if anyone adds a new print stmt. I was gonna change to use a real logger instead and have that log to a file?